### PR TITLE
some new webhooks for dear

### PR DESCRIPTION
**Pull request by kmackesy**
### <samp>🤖 Generated by Copilot at eed2e8f</samp>

This pull request adds four new source components for the DEAR Inventory app that emit events based on different webhook types. The components use a common base component for webhooks and specify the relevant metadata and version. The new components are `available-stock-level-change`, `customer-updated`, `sale-shipment-tracking-number-changed`, and `supplier-updated`.

### <samp>🤖 Generated by Copilot at eed2e8f</samp>

> _`DEAR Inventory`_
> _Webhook sources multiply_
> _Autumn harvest time_

### <samp>🤖 Generated by Copilot at eed2e8f</samp>

*  Add four new source components for the DEAR Inventory app that emit events based on different webhook types ([link](https://github.com/PipedreamHQ/pipedream/pull/6001/files?diff=unified&w=0#diff-a2d0f66b76b39b55eaa89dab760e0e9f6fca5f4ec988361c8a814161c3d7d14fR1-R32), [link](https://github.com/PipedreamHQ/pipedream/pull/6001/files?diff=unified&w=0#diff-bad7247ca747f662d633816fd34e84dd8f167cc43a9913c658bf7c18fda6cc9fR1-R32), [link](https://github.com/PipedreamHQ/pipedream/pull/6001/files?diff=unified&w=0#diff-2e969812d1707f438ce5dc82054017b677783ccf8fc3370d099521446ba5e72cR1-R32), [link](https://github.com/PipedreamHQ/pipedream/pull/6001/files?diff=unified&w=0#diff-f6dd5022400c5cffb4b0470856e3876ee442dc1d4ff9c3206aeb3380ef2a683fR1-R32))

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/dear/sources/available-stock-level-change/available-stock-level-change.mjs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import constants from "../../common/constants.mjs";
+import base from "../common/webhooks.mjs";
+
+export default {
+  ...base,
+  name: "Available Stock Level Change",
+  key: "dear-available-stock-level-change",
+  type: "source",
+  description: "Emit new event when available stock level is changed.",
```
Comment (dylburger) [inline components/dear/sources/available-stock-level-change/available-stock-level-change.mjs:9]:
```suggestion
  description: "Emit a new event when the available stock level changes.",
```

Comment (dylburger) [reply components/dear/sources/available-stock-level-change/available-stock-level-change.mjs:9]: Changed "Emit new event" to "Emit a new event" to follow Google's style guide on articles.

File: components/dear/sources/customer-updated/customer-updated.mjs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import constants from "../../common/constants.mjs";
+import base from "../common/webhooks.mjs";
+
+export default {
+  ...base,
+  name: "Customer Updated",
+  key: "dear-customer-updated",
+  type: "source",
+  description: "Emit new event when a customer is updated.",
```
Comment (dylburger) [inline components/dear/sources/customer-updated/customer-updated.mjs:9]:
```suggestion
  description: "Emit a new event when a customer is updated.",
```

Comment (dylburger) [reply components/dear/sources/customer-updated/customer-updated.mjs:9]: Changed "Emit new event" to "Emit a new event" to follow Google's style guide on articles.

File: components/dear/sources/sale-shipment-tracking-number-changed/sale-shipment-tracking-number-changed.mjs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import constants from "../../common/constants.mjs";
+import base from "../common/webhooks.mjs";
+
+export default {
+  ...base,
+  name: "Shipment Tracking Number Change",
+  key: "dear-sale-shipment-tracking-number-change",
+  type: "source",
+  description: "Emit new event when a shipment tracking number changes.",
```
Comment (dylburger) [inline components/dear/sources/sale-shipment-tracking-number-changed/sale-shipment-tracking-number-changed.mjs:9]:
```suggestion
  description: "Emit a new event when a shipment tracking number changes.",
```

Comment (dylburger) [reply components/dear/sources/sale-shipment-tracking-number-changed/sale-shipment-tracking-number-changed.mjs:9]: Changed "Emit new event" to "Emit a new event" to follow Google's style guide on articles.

File: components/dear/sources/supplier-updated/supplier-updated.mjs [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import constants from "../../common/constants.mjs";
+import base from "../common/webhooks.mjs";
+
+export default {
+  ...base,
+  name: "Supplier Updated",
+  key: "dear-supplier-updated",
+  type: "source",
+  description: "Emit new event when a supplier is updated.",
```
Comment (dylburger) [inline components/dear/sources/supplier-updated/supplier-updated.mjs:9]:
```suggestion
  description: "Emit a new event when a supplier is updated.",
```

**Review by vellames — APPROVED** [review]
lgtm

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[DEAR_Systems_6001_1115.pdf](https://github.com/PipedreamHQ/pipedream/files/11228708/DEAR_Systems_6001_1115.pdf)

**Comment by dannyroosevelt** [comment]
> Hello everyone, I have tested this PR and there're some test cases failed or needed improvement.
> 
> Please check the test report below for more information [DEAR_Systems_6001_1115.pdf](https://github.com/PipedreamHQ/pipedream/files/11228708/DEAR_Systems_6001_1115.pdf)

@vellames can you check out Leo's feedback and get this PR wrapped up?

**Comment by kmackesy** [comment]
Hi,

This will need some work, and I just haven’t had the time. I have to think through the dedupe logic on the stock updates. The customer and supplier updates are a documentation issue. I didn’t need those as quickly, so I’ve been busy trying to get other code finished. I’m happy to get back to it, but I will be a month or so.

Ken.

**Comment by vellames** [comment]
> > Hello everyone, I have tested this PR and there're some test cases failed or needed improvement.
> > Please check the test report below for more information [DEAR_Systems_6001_1115.pdf](https://github.com/PipedreamHQ/pipedream/files/11228708/DEAR_Systems_6001_1115.pdf)
> 
> @vellames can you check out Leo's feedback and get this PR wrapped up?

Sure!

**Comment by vellames** [comment]
Please use #6219 to continue with this PR. I am closing this one

**Pull request closed without merging**